# Analysis: 

In [2]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
import re
import csv

In [6]:
movies = pd.read_csv('movies.csv')
movies.head()

year  rank                                         release      gross  \
0  1980     1  Star Wars: Episode V - The Empire Strikes Back  203359628   
1  1980     2                                          9 to 5  103290500   
2  1980     3                                      Stir Crazy  101300000   
3  1980     4                               Kramer vs. Kramer   94448310   
4  1980     5                           Any Which Way You Can   70687344   

   adjusted_gross theaters  total_gross  adjusted_total_gross release_date  \
0       782934567     1278    209398025             806182396       May 21   
1       397668425      910    103290500             397668425       Dec 19   
2       390005000      813    101300000             390005000       Dec 12   
3       363625993      898    106260000             409101000       Dec 19   
4       272146274     1541     70687344             272146274       Dec 17   

             distributor release_date_formatted         seasons  
0  Twentieth Century Fox             1980-05-21          summer  
1  Twentieth Century Fox             1980-12-19  holiday season  
2      Columbia Pictures             1980-12-12  holiday season  
3      Columbia Pictures             1980-12-19  holiday season  
4           Warner Bros.             1980-12-17  holiday season

### viz 1: totals for each season -- idk bar graph or size comparison

In [7]:
movies_grouped = movies.groupby('seasons')
movies_grouped_seasons = movies_grouped['seasons'].value_counts()

movies_grouped_seasons.sort_values(ascending=False)

seasons
summer            1378
holiday season    1097
spring             634
fall               612
winter             502
Name: count, dtype: int64

seasons, total_count
summer, 1378
holiday season, 1097
spring, 635
fall, 612
winter, 502

### viz 2: seasons heatmap

In [8]:
movies_1980 = movies[movies['year'] == 1980].groupby('seasons')
movies_1980_seasons = movies_1980['seasons'].value_counts()

movies_1980_season_counts = movies_1980_seasons.sort_values(ascending=False)

print(movies_1980_season_counts)

seasons
summer            25
fall              13
holiday season    13
winter             9
spring             8
Name: count, dtype: int64


In [9]:
# rerun to reset df

header = ['year', 'summer', 'fall', 'spring', 'winter', 'holiday season']

df = pd.DataFrame(columns=header)

static test with 2018 -- SKIP

In [115]:
year = 1980

df.loc[0] = [year, movies_1980_season_counts['summer'], movies_1980_season_counts['fall'], movies_1980_season_counts['spring'], movies_1980_season_counts['winter'], movies_1980_season_counts['holiday season']]

df.astype(int)

year  summer  fall  spring  winter  holiday season
0  1980      25    13       8       9              13

let's try to loop this

In [10]:
years = range(1980, 2023)

for year in years:
    movies_year = movies[movies['year'] == year].groupby('seasons')

    movies_year_seasons = movies_year['seasons'].value_counts()

    movies_year_season_counts = movies_year_seasons.sort_values(ascending=False)

    df.loc[len(df)] = [year, movies_year_season_counts.get('summer', 0), movies_year_season_counts.get('fall', 0), movies_year_season_counts.get('spring', 0), movies_year_season_counts.get('winter', 0), movies_year_season_counts.get('holiday season', 0)]

df

year  summer  fall  spring  winter  holiday season
0   1980      25    13       8       9              13
1   1981      21     6      13       2              14
2   1982      41     8      13      11              27
3   1983      35    14      17       8              26
4   1984      28    16      17      10              29
5   1985      35    14      16       9              26
6   1986      38    12      13      13              24
7   1987      36    15      13      11              25
8   1988      34     9      19      11              27
9   1989      31    13      21      10              25
10  1990      35    17      16       7              25
11  1991      33    15      12      12              28
12  1992      31    15      17      11              26
13  1993      37    12      14      12              25
14  1994      31    13      14      15              27
15  1995      35    16      15      10              24
16  1996      33    14      13      13              27
17  1997      29    13      15      12              31
18  1998      33    15      17       9              26
19  1999      31    15      13      10              31
20  2000      33    10      16      10              31
21  2001      32    18      16      10              24
22  2002      32    12      16      14              26
23  2003      27    19      18      12              24
24  2004      33    14      19       8              26
25  2005      34    15      15      11              25
26  2006      30    12      16      14              28
27  2007      30    17      12      15              26
28  2008      29    17      15      15              24
29  2009      30    13      17      16              24
30  2010      33    14      15      13              25
31  2011      31    12      19      11              27
32  2012      33    15      13      18              21
33  2013      37    15      13      12              23
34  2014      34    18      15      10              23
35  2015      33    14      15      12              26
36  2016      37    12      12      14              25
37  2017      33    16      13      14              24
38  2018      29    16      17      13              25
39  2019      32    17      13      13              25
40  2020      17    15       5      27              35
41  2021      39    17      12       5              27
42  2022      28    19      16      10              27

In [117]:
# random checks to verify accuracy

len(movies[(movies['year'] == 2020) & (movies['seasons'] == 'summer')])

17

now let's melt this tidy data style B)

In [11]:
tidy_df = df.melt(id_vars='year', var_name='season', value_name='count')
tidy_df[tidy_df['year'] == 1980]

year          season  count
0    1980          summer     25
43   1980            fall     13
86   1980          spring      8
129  1980          winter      9
172  1980  holiday season     13

some more analysis: years where summer is not the most popular season

In [12]:
top_season = df[['summer', 'fall', 'spring', 'winter', 'holiday season']].idxmax(axis=1)

df['top_season'] = top_season

df

year  summer  fall  spring  winter  holiday season      top_season
0   1980      25    13       8       9              13          summer
1   1981      21     6      13       2              14          summer
2   1982      41     8      13      11              27          summer
3   1983      35    14      17       8              26          summer
4   1984      28    16      17      10              29  holiday season
5   1985      35    14      16       9              26          summer
6   1986      38    12      13      13              24          summer
7   1987      36    15      13      11              25          summer
8   1988      34     9      19      11              27          summer
9   1989      31    13      21      10              25          summer
10  1990      35    17      16       7              25          summer
11  1991      33    15      12      12              28          summer
12  1992      31    15      17      11              26          summer
13  1993      37    12      14      12              25          summer
14  1994      31    13      14      15              27          summer
15  1995      35    16      15      10              24          summer
16  1996      33    14      13      13              27          summer
17  1997      29    13      15      12              31  holiday season
18  1998      33    15      17       9              26          summer
19  1999      31    15      13      10              31          summer
20  2000      33    10      16      10              31          summer
21  2001      32    18      16      10              24          summer
22  2002      32    12      16      14              26          summer
23  2003      27    19      18      12              24          summer
24  2004      33    14      19       8              26          summer
25  2005      34    15      15      11              25          summer
26  2006      30    12      16      14              28          summer
27  2007      30    17      12      15              26          summer
28  2008      29    17      15      15              24          summer
29  2009      30    13      17      16              24          summer
30  2010      33    14      15      13              25          summer
31  2011      31    12      19      11              27          summer
32  2012      33    15      13      18              21          summer
33  2013      37    15      13      12              23          summer
34  2014      34    18      15      10              23          summer
35  2015      33    14      15      12              26          summer
36  2016      37    12      12      14              25          summer
37  2017      33    16      13      14              24          summer
38  2018      29    16      17      13              25          summer
39  2019      32    17      13      13              25          summer
40  2020      17    15       5      27              35  holiday season
41  2021      39    17      12       5              27          summer
42  2022      28    19      16      10              27          summer

### 

In [16]:
df.to_csv('seasons_totals.csv', index=False)
tidy_df.to_csv('seasons_totals_tidy.csv', index=False)

### viz 3: distributors preferences

In [13]:
season_counts = movies.dropna(subset=['seasons']).groupby('distributor')['seasons'].value_counts()

distributors_df = pd.DataFrame(season_counts)
distributors_df

count
distributor                       seasons              
-                                 spring             18
                                  holiday season     17
                                  summer             15
                                  fall               10
                                  winter              5
...                                                 ...
Warner Independent Pictures (WIP) summer              1
Well Go USA Entertainment         summer              2
                                  holiday season      1
Yari Film Group Releasing         summer              1
truTV                             winter              1

[279 rows x 1 columns]

top 10 distributors



In [14]:
top_distributors = movies['distributor'].value_counts().head(10)
top_distributors_list = top_distributors.index.to_list()
top_distributors_list

['Warner Bros.',
 'Universal Pictures',
 'Walt Disney Studios Motion Pictures',
 'Twentieth Century Fox',
 'Paramount Pictures',
 'Sony Pictures Entertainment (SPE)',
 'Metro-Goldwyn-Mayer (MGM)',
 'Lionsgate',
 'Columbia Pictures',
 'New Line Cinema']

In [15]:
distributors = movies[movies['distributor'].isin(top_distributors_list)]
distributors

year  rank                                         release      gross  \
0     1980     1  Star Wars: Episode V - The Empire Strikes Back  203359628   
1     1980     2                                          9 to 5  103290500   
2     1980     3                                      Stir Crazy  101300000   
3     1980     4                               Kramer vs. Kramer   94448310   
4     1980     5                           Any Which Way You Can   70687344   
...    ...   ...                                             ...        ...   
4212  2022    89                                            Fall    7240521   
4214  2022    91                         Ghostbusters: Afterlife    6981615   
4217  2022    94                                        She Said    5825995   
4220  2022    97                                         Encanto    5465293   
4221  2022    98                              Orphan: First Kill    5413227   

      adjusted_gross theaters  total_gross  adjusted_total_gross release_date  \
0          782934567     1278    209398025             806182396       May 21   
1          397668425      910    103290500             397668425       Dec 19   
2          390005000      813    101300000             390005000       Dec 12   
3          363625993      898    106260000             409101000       Dec 19   
4          272146274     1541     70687344             272146274       Dec 17   
...              ...      ...          ...                   ...          ...   
4212         7674952     1548      7240521               7674952       Aug 12   
4214         7400511     4315    129360575             137122209       Nov 19   
4217         6175554     2023      5825995               6175554       Nov 18   
4220         5793210     3980     96093622             101859239       Nov 24   
4221         5738020      557      5413227               5738020       Aug 19   

                              distributor release_date_formatted  \
0                   Twentieth Century Fox             1980-05-21   
1                   Twentieth Century Fox             1980-12-19   
2                       Columbia Pictures             1980-12-12   
3                       Columbia Pictures             1980-12-19   
4                            Warner Bros.             1980-12-17   
...                                   ...                    ...   
4212                            Lionsgate             2022-08-12   
4214    Sony Pictures Entertainment (SPE)             2022-11-19   
4217                   Universal Pictures             2022-11-18   
4220  Walt Disney Studios Motion Pictures             2022-11-24   
4221                   Paramount Pictures             2022-08-19   

             seasons  
0             summer  
1     holiday season  
2     holiday season  
3     holiday season  
4     holiday season  
...              ...  
4212          summer  
4214  holiday season  
4217  holiday season  
4220  holiday season  
4221          summer  

[3347 rows x 12 columns]

In [16]:
distributors_seasons = distributors.groupby('distributor')['seasons'].value_counts(ascending=False)

# TO RESET DF

In [17]:
distributors_seasons_df = pd.DataFrame(distributors_seasons)
distributors_seasons_df = distributors_seasons_df.reset_index()

distributors_seasons_df

distributor         seasons  count
0                     Columbia Pictures          summer     41
1                     Columbia Pictures  holiday season     39
2                     Columbia Pictures            fall     23
3                     Columbia Pictures          spring     17
4                     Columbia Pictures          winter      9
5                             Lionsgate          summer     32
6                             Lionsgate  holiday season     28
7                             Lionsgate            fall     26
8                             Lionsgate          spring     25
9                             Lionsgate          winter     20
10            Metro-Goldwyn-Mayer (MGM)          summer     42
11            Metro-Goldwyn-Mayer (MGM)          spring     26
12            Metro-Goldwyn-Mayer (MGM)  holiday season     23
13            Metro-Goldwyn-Mayer (MGM)            fall     23
14            Metro-Goldwyn-Mayer (MGM)          winter     18
15                      New Line Cinema  holiday season     30
16                      New Line Cinema          summer     25
17                      New Line Cinema          spring     25
18                      New Line Cinema            fall     18
19                      New Line Cinema          winter     16
20                   Paramount Pictures          summer    139
21                   Paramount Pictures  holiday season    119
22                   Paramount Pictures          spring     56
23                   Paramount Pictures          winter     54
24                   Paramount Pictures            fall     49
25    Sony Pictures Entertainment (SPE)          summer     99
26    Sony Pictures Entertainment (SPE)  holiday season     81
27    Sony Pictures Entertainment (SPE)            fall     43
28    Sony Pictures Entertainment (SPE)          spring     34
29    Sony Pictures Entertainment (SPE)          winter     34
30                Twentieth Century Fox          summer    154
31                Twentieth Century Fox  holiday season    135
32                Twentieth Century Fox          spring     59
33                Twentieth Century Fox          winter     49
34                Twentieth Century Fox            fall     47
35                   Universal Pictures          summer    201
36                   Universal Pictures  holiday season    132
37                   Universal Pictures          spring     91
38                   Universal Pictures            fall     72
39                   Universal Pictures          winter     66
40  Walt Disney Studios Motion Pictures          summer    165
41  Walt Disney Studios Motion Pictures  holiday season    147
42  Walt Disney Studios Motion Pictures          spring     76
43  Walt Disney Studios Motion Pictures            fall     60
44  Walt Disney Studios Motion Pictures          winter     49
45                         Warner Bros.          summer    227
46                         Warner Bros.  holiday season    152
47                         Warner Bros.            fall     92
48                         Warner Bros.          spring     87
49                         Warner Bros.          winter     71

### NVM : who's coming out on top?
average gross for each season

In [38]:
movies_test = movies[(movies['distributor'] == 'Twentieth Century Fox') & (movies['seasons'] == 'summer')]
movies_test_gross = movies_test['adjusted_gross'].max()

movies_test_gross

783113108

In [18]:
seasons = ['summer', 'fall', 'spring', 'winter', 'holiday season']

for distributor in top_distributors_list:
    for season in seasons:
        movies_filtered = movies[(movies['distributor'] == distributor) & (movies['seasons'] == season)]
        movies_gross = movies_filtered['adjusted_gross'].mean()
        distributors_seasons_df.loc[
            (distributors_seasons_df['distributor'] == distributor) & (distributors_seasons_df['seasons'] == season),
            'average_adjusted_gross'
        ] = round(movies_gross, 0)

distributors_seasons_df['average_adjusted_gross'] = distributors_seasons_df['average_adjusted_gross'].astype(int)
distributors_seasons_df

distributor         seasons  count  \
0                     Columbia Pictures          summer     41   
1                     Columbia Pictures  holiday season     39   
2                     Columbia Pictures            fall     23   
3                     Columbia Pictures          spring     17   
4                     Columbia Pictures          winter      9   
5                             Lionsgate          summer     32   
6                             Lionsgate  holiday season     28   
7                             Lionsgate            fall     26   
8                             Lionsgate          spring     25   
9                             Lionsgate          winter     20   
10            Metro-Goldwyn-Mayer (MGM)          summer     42   
11            Metro-Goldwyn-Mayer (MGM)          spring     26   
12            Metro-Goldwyn-Mayer (MGM)  holiday season     23   
13            Metro-Goldwyn-Mayer (MGM)            fall     23   
14            Metro-Goldwyn-Mayer (MGM)          winter     18   
15                      New Line Cinema  holiday season     30   
16                      New Line Cinema          summer     25   
17                      New Line Cinema          spring     25   
18                      New Line Cinema            fall     18   
19                      New Line Cinema          winter     16   
20                   Paramount Pictures          summer    139   
21                   Paramount Pictures  holiday season    119   
22                   Paramount Pictures          spring     56   
23                   Paramount Pictures          winter     54   
24                   Paramount Pictures            fall     49   
25    Sony Pictures Entertainment (SPE)          summer     99   
26    Sony Pictures Entertainment (SPE)  holiday season     81   
27    Sony Pictures Entertainment (SPE)            fall     43   
28    Sony Pictures Entertainment (SPE)          spring     34   
29    Sony Pictures Entertainment (SPE)          winter     34   
30                Twentieth Century Fox          summer    154   
31                Twentieth Century Fox  holiday season    135   
32                Twentieth Century Fox          spring     59   
33                Twentieth Century Fox          winter     49   
34                Twentieth Century Fox            fall     47   
35                   Universal Pictures          summer    201   
36                   Universal Pictures  holiday season    132   
37                   Universal Pictures          spring     91   
38                   Universal Pictures            fall     72   
39                   Universal Pictures          winter     66   
40  Walt Disney Studios Motion Pictures          summer    165   
41  Walt Disney Studios Motion Pictures  holiday season    147   
42  Walt Disney Studios Motion Pictures          spring     76   
43  Walt Disney Studios Motion Pictures            fall     60   
44  Walt Disney Studios Motion Pictures          winter     49   
45                         Warner Bros.          summer    227   
46                         Warner Bros.  holiday season    152   
47                         Warner Bros.            fall     92   
48                         Warner Bros.          spring     87   
49                         Warner Bros.          winter     71   

    average_adjusted_gross  
0                125098311  
1                 94398449  
2                 58599908  
3                 44459054  
4                 49592802  
5                 71078150  
6                104033627  
7                 67098261  
8                 85382735  
9                 63013881  
10                82945611  
11                59759606  
12                98854813  
13                67821657  
14                50734336  
15               116491807  
16               141792065  
17                79252739  
18                73282136  
19                72825496  
20               156777431  
21                93419

top + least grossing film of each season for each distributor

In [46]:
# disney_movies = movies[movies['distributor'] == 'Walt Disney Studios Motion Pictures']
# disney_movies_seasons = disney_movies.groupby('seasons')

# # find index of the row with the highest 'gross' value for each season
# disney_top_grossing_index = disney_movies_seasons['adjusted_gross'].idxmax()

# disney_top_grossing_movies = disney_movies.loc[disney_top_grossing_index]

# disney_top_grossing_movies[['seasons', 'adjusted_gross', 'release', 'year']]

seasons  adjusted_gross  \
2026            fall       406067277   
3425  holiday season       834518104   
3824          spring      1021463870   
3726          summer       736383910   
3724          winter       847072074   

                                         release  year  
2026                              Monsters, Inc.  2001  
3425  Star Wars: Episode VII - The Force Awakens  2015  
3824                           Avengers: Endgame  2019  
3726                               Incredibles 2  2018  
3724                               Black Panther  2018

In [184]:
# movies[(movies['distributor'] == 'Walt Disney Studios Motion Pictures') & (movies['seasons'] == 'summer')].sort_values(by='gross', ascending = False)

### top film for each season + year released

In [51]:
# distributors_seasons_df['top_release'] = pd.NA
# distributors_seasons_df['top_year'] = pd.NA
# distributors_seasons_df['top_gross'] = pd.NA

# for distributor in top_distributors_list:
#     for season in seasons:
#         distributor_movies = movies[(movies['distributor'] == distributor) & (movies['seasons'] == season)]

#         if not distributor_movies.empty:
#             top_grossing_index = distributor_movies['adjusted_gross'].idxmax()
#             top_grossing_release = distributor_movies.loc[top_grossing_index, 'release']
#             top_grossing_year = distributor_movies.loc[top_grossing_index, 'year']
#             top_grossing_gross = distributor_movies.loc[top_grossing_index, 'adjusted_gross']
            

#             distributors_seasons_df.loc[(distributors_seasons_df['distributor'] == distributor) &
#                                         (distributors_seasons_df['seasons'] == season),
#                                         'top_release'] = top_grossing_release

#             distributors_seasons_df.loc[(distributors_seasons_df['distributor'] == distributor) &
#                                         (distributors_seasons_df['seasons'] == season),
#                                         'top_year'] = top_grossing_year
            
#             distributors_seasons_df.loc[(distributors_seasons_df['distributor'] == distributor) &
#                                         (distributors_seasons_df['seasons'] == season),
#                                         'top_gross'] = top_grossing_gross
                                

# distributors_seasons_df

distributor         seasons  count  \
0                     Columbia Pictures          summer     41   
1                     Columbia Pictures  holiday season     39   
2                     Columbia Pictures            fall     23   
3                     Columbia Pictures          spring     17   
4                     Columbia Pictures          winter      9   
5                             Lionsgate          summer     32   
6                             Lionsgate  holiday season     28   
7                             Lionsgate            fall     26   
8                             Lionsgate          spring     25   
9                             Lionsgate          winter     20   
10            Metro-Goldwyn-Mayer (MGM)          summer     42   
11            Metro-Goldwyn-Mayer (MGM)          spring     26   
12            Metro-Goldwyn-Mayer (MGM)  holiday season     23   
13            Metro-Goldwyn-Mayer (MGM)            fall     23   
14            Metro-Goldwyn-Mayer (MGM)          winter     18   
15                      New Line Cinema  holiday season     30   
16                      New Line Cinema          summer     25   
17                      New Line Cinema          spring     25   
18                      New Line Cinema            fall     18   
19                      New Line Cinema          winter     16   
20                   Paramount Pictures          summer    139   
21                   Paramount Pictures  holiday season    119   
22                   Paramount Pictures          spring     56   
23                   Paramount Pictures          winter     54   
24                   Paramount Pictures            fall     49   
25    Sony Pictures Entertainment (SPE)          summer     99   
26    Sony Pictures Entertainment (SPE)  holiday season     81   
27    Sony Pictures Entertainment (SPE)            fall     43   
28    Sony Pictures Entertainment (SPE)          spring     34   
29    Sony Pictures Entertainment (SPE)          winter     34   
30                Twentieth Century Fox          summer    154   
31                Twentieth Century Fox  holiday season    135   
32                Twentieth Century Fox          spring     59   
33                Twentieth Century Fox          winter     49   
34                Twentieth Century Fox            fall     47   
35                   Universal Pictures          summer    201   
36                   Universal Pictures  holiday season    132   
37                   Universal Pictures          spring     91   
38                   Universal Pictures            fall     72   
39                   Universal Pictures          winter     66   
40  Walt Disney Studios Motion Pictures          summer    165   
41  Walt Disney Studios Motion Pictures  holiday season    147   
42  Walt Disney Studios Motion Pictures          spring     76   
43  Walt Disney Studios Motion Pictures            fall     60   
44  Walt Disney Studios Motion Pictures          winter     49   
45                         Warner Bros.          summer    227   
46                         Warner Bros.  holiday season    152   
47                         Warner Bros.            fall     92   
48                         Warner Bros.          spring     87   
49                         Warner Bros.          winter     71   

    average_adjusted_gross                                    top_release  \
0                125098311                                   Ghostbusters   
1                 94398449                                        Tootsie   
2                 58599908                               Chariots of Fire   
3                 44459054        Richard Pryor: Live on the Sunset Strip   
4                 49592802                                  Groundhog Day   
5                 71078150              John Wick: Chapter 3 - Parabellum   
6                104033627                The Hunger Games: Catching Fire   
7                 67098261                   

# actually let's redo that

In [21]:
movies_gross = pd.read_csv('movies_gross.csv')
movies_gross.head()


release                          distributor  \
0                         '83               Reliance Entertainment   
1     *batteries not included                   Universal Pictures   
2         10 Cloverfield Lane                   Paramount Pictures   
3  10 Things I Hate About You  Walt Disney Studios Motion Pictures   
4              10 to Midnight            Metro-Goldwyn-Mayer (MGM)   

          seasons  adjusted_gross  
0  holiday season         3448423  
1  holiday season        86795885  
2          spring        90824577  
3          spring        69484262  
4          spring        21957311

In [22]:
distributors_seasons_df['top_release'] = pd.NA
distributors_seasons_df['top_gross'] = pd.NA

for distributor in top_distributors_list:
    for season in seasons:
        distributor_movies = movies_gross[(movies_gross['distributor'] == distributor) & (movies_gross['seasons'] == season)]

        if not distributor_movies.empty:
            top_grossing_index = distributor_movies['adjusted_gross'].idxmax()
            top_grossing_release = distributor_movies.loc[top_grossing_index, 'release']
            top_grossing_gross = distributor_movies.loc[top_grossing_index, 'adjusted_gross']
            

            distributors_seasons_df.loc[(distributors_seasons_df['distributor'] == distributor) &
                                        (distributors_seasons_df['seasons'] == season),
                                        'top_release'] = top_grossing_release
            
            distributors_seasons_df.loc[(distributors_seasons_df['distributor'] == distributor) &
                                        (distributors_seasons_df['seasons'] == season),
                                        'top_gross'] = top_grossing_gross
                                

distributors_seasons_df

distributor         seasons  count  \
0                     Columbia Pictures          summer     41   
1                     Columbia Pictures  holiday season     39   
2                     Columbia Pictures            fall     23   
3                     Columbia Pictures          spring     17   
4                     Columbia Pictures          winter      9   
5                             Lionsgate          summer     32   
6                             Lionsgate  holiday season     28   
7                             Lionsgate            fall     26   
8                             Lionsgate          spring     25   
9                             Lionsgate          winter     20   
10            Metro-Goldwyn-Mayer (MGM)          summer     42   
11            Metro-Goldwyn-Mayer (MGM)          spring     26   
12            Metro-Goldwyn-Mayer (MGM)  holiday season     23   
13            Metro-Goldwyn-Mayer (MGM)            fall     23   
14            Metro-Goldwyn-Mayer (MGM)          winter     18   
15                      New Line Cinema  holiday season     30   
16                      New Line Cinema          summer     25   
17                      New Line Cinema          spring     25   
18                      New Line Cinema            fall     18   
19                      New Line Cinema          winter     16   
20                   Paramount Pictures          summer    139   
21                   Paramount Pictures  holiday season    119   
22                   Paramount Pictures          spring     56   
23                   Paramount Pictures          winter     54   
24                   Paramount Pictures            fall     49   
25    Sony Pictures Entertainment (SPE)          summer     99   
26    Sony Pictures Entertainment (SPE)  holiday season     81   
27    Sony Pictures Entertainment (SPE)            fall     43   
28    Sony Pictures Entertainment (SPE)          spring     34   
29    Sony Pictures Entertainment (SPE)          winter     34   
30                Twentieth Century Fox          summer    154   
31                Twentieth Century Fox  holiday season    135   
32                Twentieth Century Fox          spring     59   
33                Twentieth Century Fox          winter     49   
34                Twentieth Century Fox            fall     47   
35                   Universal Pictures          summer    201   
36                   Universal Pictures  holiday season    132   
37                   Universal Pictures          spring     91   
38                   Universal Pictures            fall     72   
39                   Universal Pictures          winter     66   
40  Walt Disney Studios Motion Pictures          summer    165   
41  Walt Disney Studios Motion Pictures  holiday season    147   
42  Walt Disney Studios Motion Pictures          spring     76   
43  Walt Disney Studios Motion Pictures            fall     60   
44  Walt Disney Studios Motion Pictures          winter     49   
45                         Warner Bros.          summer    227   
46                         Warner Bros.  holiday season    152   
47                         Warner Bros.            fall     92   
48                         Warner Bros.          spring     87   
49                         Warner Bros.          winter     71   

    average_adjusted_gross                                    top_release  \
0                125098311                                   Ghostbusters   
1                 94398449                                        Tootsie   
2                 58599908                               Chariots of Fire   
3                 44459054        Richard Pryor: Live on the Sunset Strip   
4                 49592802                                  Groundhog Day   
5                 71078150              John Wick: Chapter 3 - Parabellum   
6                104033627                The Hunger Games: Catching Fire   
7                 67098261                   

In [23]:
distributors_seasons_df.to_csv('distributors_seasons.csv', index=False)

In [26]:
movies[(movies['distributor'] == 'Walt Disney Studios Motion Pictures') & (movies['seasons'] == 'summer')].sort_values(by='adjusted_gross', ascending = False)

year  rank                                            release  \
3726  2018     3                                      Incredibles 2   
3825  2019     2                                      The Lion King   
2524  2006     1         Pirates of the Caribbean: Dead Man's Chest   
3524  2016     1                                       Finding Dory   
1324  1994     1                                      The Lion King   
...    ...   ...                                                ...   
504   1985    81                                       Return to Oz   
1520  1995    97                                           Mad Love   
1219  1992    96                                A Stranger Among Us   
920   1989    97                                            Cheetah   
3990  2020    67  Star Wars: Episode V - The Empire Strikes Back...   

          gross  adjusted_gross theaters  total_gross  adjusted_total_gross  \
3726  608581744       736383910     4410    608581744             736383910   
3825  543638043       646929271     4802    543638043             646929271   
2524  423317154       639208902     4133    423315812             639206876   
3524  486295561       612732406     4305    486295561             612732406   
1324  295691076       606166705     2624    312855561             641353900   
...         ...             ...      ...          ...                   ...   
504    11137801        31631354     1238     11137801              31631354   
1520   15453274        30752015     1500     15453274              30752015   
1219   12282994        26654096      684     12282994              26654096   
920     8153677        20139582     1321      8153677              20139582   
3990    2482000         2879120     2097      2482000               2879120   

     release_date                          distributor release_date_formatted  \
3726       Jun 15  Walt Disney Studios Motion Pictures             2018-06-15   
3825       Jul 19  Walt Disney Studios Motion Pictures             2019-07-19   
2524        Jul 7  Walt Disney Studios Motion Pictures             2006-07-07   
3524       Jun 17  Walt Disney Studios Motion Pictures             2016-06-17   
1324       Jun 15  Walt Disney Studios Motion Pictures             1994-06-15   
...           ...                                  ...                    ...   
504        Jun 21  Walt Disney Studios Motion Pictures             1985-06-21   
1520       May 26  Walt Disney Studios Motion Pictures             1995-05-26   
1219       Jul 17  Walt Disney Studios Motion Pictures             1992-07-17   
920        Aug 18  Walt Disney Studios Motion Pictures             1989-08-18   
3990       Jul 10  Walt Disney Studios Motion Pictures             2020-07-10   

     seasons  
3726  summer  
3825  summer  
2524  summer  
3524  summer  
1324  summer  
...      ...  
504   summer  
1520  summer  
1219  summer  
920   summer  
3990  summer  

[165 rows x 12 columns]